produce.go

```diff
@@ -320,11 +320,16 @@ func (cmd *produceCmd) deserializeLines(in chan string, out chan message, partit
 			}
 
 			var part int32 = 0
-			if msg.Key != nil && cmd.partitioner == "hashCode" {
-				part = hashCodePartition(*msg.Key, partitionCount)
-			}
-			if msg.Partition == nil {
+			if msg.Value != nil && cmd.partitioner == "hashCodeByValue" {
+				part = hashCodePartition(*msg.Value, partitionCount)
 				msg.Partition = &part
+			}else {
```

Comment:
could you please run gofmt or your alternative on this? 🙏

Reply:
and would it make sense to group the guards slightly differently?

```go
if msg.Partition == nil {
	if msg.Value != nil && cmd.partitioner == "hashCodeByValue" {
		part = hashCodePartition(*msg.Value, partitionCount)
		msg.Partition = &part
	} else if msg.Key != nil && cmd.partitioner == "hashCode" {
		part = hashCodePartition(*msg.Key, partitionCount)
		msg.Partition = &part
	}
}
```

Reply:
ok, I will do it